packages/components/inline-message/src/inline-message.ts

```diff
@@ -0,0 +1,119 @@
+import type { TemplateResult } from 'lit-html';
+import type { CSSResultGroup, PropertyValues } from 'lit';
+import type { ScopedElementsMap } from '@open-wc/scoped-elements/lit-element.js';
+import { ScopedElementsMixin } from '@open-wc/scoped-elements/lit-element.js';
+import { Icon } from '@sl-design-system/icon';
+import { Button } from '@sl-design-system/button';
+import { breakpoints } from '@sl-design-system/shared';
+import { localized, msg } from '@lit/localize';
+import { LitElement, html, nothing } from 'lit';
+import { property, query } from 'lit/decorators.js';
+import styles from './inline-message.scss.js';
+
+export type InlineMessageStatus = 'info' | 'success' | 'warning' | 'danger';
+
+/**
+ * An inline message component for displaying additional information/errors.
+ *
+ * @slot default - title content for the inline message
+ * @slot description - slot for additional information and more content for the inline-message
+ * @slot details - slot for more details of the inline-message like list of errors
+ * @slot icon - icon shown on the left side of the component
+ * @slot close-button - Closing button for the inline message
+ */
+@localized()
+export class InlineMessage extends ScopedElementsMixin(LitElement) {
+  /** @private */
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-button': Button,
+      'sl-icon': Icon
+    };
+  }
+
+  /** @private */
+  static override styles: CSSResultGroup = [breakpoints, styles];
+
+  /** @private */
+  @query('.wrapper') wrapper?: HTMLDivElement;
+
+  /** Determines whether closing button (default one) should be shown in the top right corner. */
+  @property({ type: Boolean, reflect: true }) dismissible = true;
+
+  /** Determines whether the icon should be shown on the left side of the component. */
+  @property({ type: Boolean, attribute: 'no-icon' }) noIcon?: boolean;
+
+  /** The status of the inline message.
+   * @type {'info' | 'success' | 'warning' | 'danger'} */
+  @property({ reflect: true }) status: InlineMessageStatus = 'info';
```

Comment:
I'm not sure "status" is the best term. I'd use "variant" just like we do for the buttons.

Reply:
I used `status` to have it the same like in figma
<img width="228" alt="image" src="https://github.com/sl-design-system/components/assets/111562742/0dc5fac3-4db4-408a-8bbb-f52cd910f3f2">

Reply:
We also have status in acceptance criteria, so maybe we should leave `status` for now?
https://github.com/sl-design-system/components/issues/277